File: python/matrix/matrix.py

class Matrix:
    def __init__(self, matrix_string):
        self.__matrix = []
        rows = matrix_string.split('\n')
        for row in rows:
            numbers = row.split(' ')
            numbers = list(map(int, numbers))
            self.__matrix.append(numbers)
        print(self.__matrix)

    def row(self, index):
        return self.__matrix[index - 1]

    def column(self, index):
        return list(map(lambda row: row[index - 1], self.__matrix))

File: ruby/matrix/matrix.rb
=begin
Write your code for the 'Matrix' exercise in this file. Make the tests in
`matrix_test.rb` pass.

To get started with TDD, see the `README.md` file in your
`ruby/matrix` directory.
=end

class Matrix

    attr_reader :rows

    def initialize(input)
        @rows = input.split("\n").map do |row|
            row.split(' ').map(&:to_i)
        end
    end

    def columns
        @rows.transpose
    end

    def saddle_points
      # code here
    end
end

